realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -1209,6 +1211,70 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
         Realm.migrateRealm(config, migration);
     }
 
+    @Test
+    public void renameAndAddField() {
+        RealmMigration migration = new RealmMigration() {
+            @Override
+            public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
+                realm.schema.get("MigrationFieldRenameAndAdd")
+                        .renameField("string1", "string2")
+                        .addField("string1", String.class);
+            }
+        };
+
+        RealmConfiguration config = configFactory.createConfigurationBuilder()
+                .schema(MigrationFieldRenameAndAdd.class)
+                .schemaVersion(2)
+                .migration(migration)
+                .assetFile("rename-and-add.realm")
+                .build();
+
+        Realm realm = Realm.getInstance(config);
+        assertTrue(realm.getSchema().get("MigrationFieldRenameAndAdd").hasField("string1"));
+        assertTrue(realm.getSchema().get("MigrationFieldRenameAndAdd").hasField("string2"));
+        realm.close();
+    }
+
+    @Test
+    public void renameAndAddIndexedField() {
+        final int oldTestVal = 7;
+        final Long testVal = Long.valueOf(293);
+
+        RealmMigration migration = new RealmMigration() {
+            @Override
+            public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
+                realm.schema.get("MigrationIndexedFieldRenamed") // WTF??  NO SUCH SCHEMA?
+                        .renameField("testField", "oldTestField")
+                        .addField("testField", Long.class);
+            }
+        };
+
+        RealmConfiguration config = configFactory.createConfigurationBuilder()
+                .schema(MigrationIndexedFieldRenamed.class)
+                .schemaVersion(2)
+                .migration(migration)
+                .assetFile("rename-and-add-indexed.realm")
+                .build();
+        realm = Realm.getInstance(config);
+
+        realm.beginTransaction();
+        MigrationIndexedFieldRenamed obj = realm.createObject(MigrationIndexedFieldRenamed.class, 2);
+        obj.oldTestField = oldTestVal;
+        obj.testField = testVal;
+        realm.commitTransaction();
+
+        assertTrue(realm.getSchema().get("MigrationIndexedFieldRenamed").hasField("testField"));
+        assertTrue(realm.getSchema().get("MigrationIndexedFieldRenamed").hasField("oldTestField"));
```

Comment:
Maybe check if index is still preserved. Since the original issue is about wrong number of columns, it might be worth to check that too.